pkg/apic/apiservicerevision.go

```diff
@@ -102,52 +102,60 @@ func (c *ServiceClient) processRevision(serviceBody *ServiceBody) error {
 
 // GetAPIRevisions - Returns the list of API revisions for the specified filter
 func (c *ServiceClient) GetAPIRevisions(queryParams map[string]string, stage string) ([]v1alpha1.APIServiceRevision, error) {
+	//TODO - SHANE need to check with mulesoft for this
 	return c.getAPIRevisions(queryParams, stage)
 }
 
 // getAPIRevisions - Returns the list of API revisions for the specified filter
 func (c *ServiceClient) getAPIRevisions(queryParams map[string]string, stage string) ([]v1alpha1.APIServiceRevision, error) {
-	headers, err := c.createHeader()
-	if err != nil {
-		return nil, err
-	}
+	apiRevisionsURL := c.cfg.GetRevisionsURL()
+	morePages := true
+	page := 1
 
-	request := coreapi.Request{
-		Method:      coreapi.GET,
-		URL:         c.cfg.GetRevisionsURL(),
-		Headers:     headers,
-		QueryParams: queryParams,
-	}
+	apiRevisions := make([]v1alpha1.APIServiceRevision, 0)
+	filteredAPIRevisions := make([]v1alpha1.APIServiceRevision, 0)
 
-	response, err := c.apiClient.Send(request)
-	if err != nil {
-		return nil, err
-	}
-	if response.Code != http.StatusOK {
-		if response.Code != http.StatusNotFound {
-			responseErr := readResponseErrors(response.Code, response.Body)
-			return nil, utilerrors.Wrap(ErrRequestQuery, responseErr)
+	for morePages {
```

Comment:
This is not supposed to use GetAPIServiceRevisions or similar?

Reply:
No @jcollins-axway  I did not change this specifically.  Left this one in there for @tsjohns9 to look at.  Because the calling func is coming from Mulesoft I'm assuming since @tsjohns9 created a GetAPIRevisions func

`// GetAPIRevisions - Returns the list of API revisions for the specified filter
func (c *ServiceClient) GetAPIRevisions(queryParams map[string]string, stage string) ([]v1alpha1.APIServiceRevision, error) {
	return c.getAPIRevisions(queryParams, stage)
}`

I tagged him to look at it and see if he can leverage the func in paginationResource like the others are